tests/test_progress.py

```
@@ -0,0 +1,251 @@
+
+    def setUp(self):
+        """Remember the current switch state."""
+        self._original = progress._DISABLED
+        progress._DISABLED = False
+
+    def tearDown(self):
+        """Restore the original switch state."""
+        progress._DISABLED = self._original
+
+    @contextlib.contextmanager
+    def _record_bars(self):
```

Ruff (0.15.21)

[warning] Missing return type annotation for private function `_record_bars`

(ANN202)